api/src/main/java/org/openmrs/module/fgh/mpi/miscellaneous/TokenInfo.java

```diff
@@ -0,0 +1,79 @@
+package org.openmrs.module.fgh.mpi.miscellaneous;
+
+import org.apache.commons.lang.time.StopWatch;
+
+public class TokenInfo {
+	
+	private String access_token;
+	
+	private String id_token;
+	
+	private String token_type;
+	
+	private Double expires_in;
+	
+	private String refresh_token;
+	
+	private StopWatch stopWach;
+	
+	public TokenInfo() {
+	}
+	
+	public String getAccess_token() {
+		return access_token;
+	}
+	
+	public void setAccess_token(String access_token) {
+		this.access_token = access_token;
+	}
+	
+	public String getId_token() {
+		return id_token;
+	}
+	
+	public void setId_token(String id_token) {
+		this.id_token = id_token;
+	}
+	
+	public String getToken_type() {
+		return token_type;
+	}
+	
+	public void setToken_type(String token_type) {
+		this.token_type = token_type;
+	}
+	
+	public Double getExpires_in() {
+		return expires_in;
+	}
+	
+	public void setExpires_in(Double expires_in) {
+		this.expires_in = expires_in;
+	}
+	
+	public String getRefresh_token() {
+		return refresh_token;
+	}
+	
+	public void setRefresh_token(String refresh_token) {
+		this.refresh_token = refresh_token;
+	}
+	
+	public boolean isValid() {
```

Comment:
It would be nice to include tests for this logic too

Reply:
Created a new class TokenInfoTest with the test.